pkg/client/connection.go

```diff
@@ -51,21 +61,24 @@ func CreateApiConnectionWithCallOptions(
 
 	callOptions := append(additionalDefaultCallOptions, grpc.WaitForReady(true))
 	defaultCallOptions := grpc.WithDefaultCallOptions(callOptions...)
-	keepAliveOptions := grpc.WithKeepaliveParams(
-		keepalive.ClientParameters{
-			Time:    15 * time.Second,
-			Timeout: 10 * time.Second,
-		},
-	)
 	unuaryInterceptors := grpc.WithChainUnaryInterceptor(grpc_retry.UnaryClientInterceptor(retryOpts...))
 	streamInterceptors := grpc.WithChainStreamInterceptor(grpc_retry.StreamClientInterceptor(retryOpts...))
 	dialOpts := append(additionalDialOptions,
 		defaultCallOptions,
-		keepAliveOptions,
 		unuaryInterceptors,
 		streamInterceptors,
 		transportCredentials(config),
 	)
+	// gRPC keepalive options.
+	if config.GrpcKeepAliveTime > 0 || config.GrpcKeepAliveTimeout > 0 {
```

Comment:
is this correct? What happens if `config.GrpcKeepAliveTime` = 0 and  `config.GrpcKeepAliveTime` = 1.  Should it be `&&`?

Reply:
I didn't want to be opinionated here on which settings to allow. I prefer to put in known-good default values later once we've rolled out a server that supports frequent keepalive everywhere.